### PR TITLE
Adapt to new haskell-src-exts API.

### PR DESCRIPTION
**Pull request by eigengrau**
VarA has been refactored into AppA as of HSE a4feb22.

### CODE REVIEW

File: src/HIndent/Pretty.hs [hunk 1]
```diff
@@ -922,7 +922,7 @@ instance Pretty Asst where
            write " ~ "
            pretty b
       ParenA _ asst -> parens $ pretty asst
-      VarA _ var -> pretty var
+      AppA _ var _ -> pretty var
```
Comment (mpickering) [inline src/HIndent/Pretty.hs]: I don't think this can be right. Surely the third argument also needs to be printed.

**Comment by eigengrau** [comment]
@mpickering: Right, I should’ve had a look at the semantics. I was a bit rushed to get my tools sandbox to build with HSE’s head. I amended based on what’s done with `ClassA`.

Out of curiosity: I noticed the HSE constructors have three arguments when used here, but only two when defined in the HSE sources. Does this have to do with `deriving Generic`?

**Comment by eigengrau** [comment]
While I was at it, I added some FIXMEs for inexhaustive pattern matches (new HSE constructors?).

**Comment by mpickering** [comment]
HSE has two syntax trees, an annotated one which includes more source locations and an unannotated one which doesn't have the source locations. You are probably looking at the unannotated one rather than the annotated one.

Yes, there are some new constructors to deal with type and expression wildcards.

**Comment by chrisdone** [comment]
I think we're good now.

**Pull request closed without merging**